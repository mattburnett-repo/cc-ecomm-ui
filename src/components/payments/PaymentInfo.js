export default function PaymentInfo () {
    return (
        <div>
            <div>
                <label htmlFor="name-on-card">Name on Card:</label>
                <input id="name-on-card" aria-label="name-on-card" />
            </div>
            <div>
                <label htmlFor="card-number">Card Number:</label>
                <input id="card-number" aria-label="card-number" />
            </div>
            <div>
                <label htmlFor="expiration-date">Exp. Date:</label>
                <input id="xpiration-date" aria-label="expiration-date" />
            </div>
            <button aria-label="finish-order">Finish Order</button>
        </div>
    )
}