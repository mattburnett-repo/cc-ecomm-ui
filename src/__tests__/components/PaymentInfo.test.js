// 20211217: Wireframe https://wireframe.cc/nx8Dv9

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

import PaymentInfo from '../../components/payments/PaymentInfo';

describe('<PaymentInfo /> component tests', () => {
    beforeEach(() => render(<PaymentInfo />));

    it('should display name on card', () => {
        screen.getByRole('textbox', {name: /name-on-card/i});
    });
    it('should display card number', () => {
        screen.getByRole('textbox', {name: /card-number/i});
    });
    it('should display expiration date', () => {
        screen.getByRole('textbox', {name: /expiration-date/i});
    });
    it('should display finish order button', () => {
        screen.getByRole('button', {name: /finish-order/i});
    });
}); // end component

describe('<PaymentInfo /> feature tests', () => {
    beforeEach(() => render(<PaymentInfo />));

    it('accepts name on card', () => {
        const theElement = screen.getByRole('textbox', {name: /name-on-card/i});
        userEvent.type(theElement, 'name-on-card');
        
        expect(theElement).toHaveValue('name-on-card');
    });
    it('accepts card number', () => {
        const theElement = screen.getByRole('textbox', {name: /card-number/i});
        userEvent.type(theElement, '1234 1234 1234 1234');
        
        expect(theElement).toHaveValue('1234 1234 1234 1234');
    });
    it('accepts expiration date', () => {
        const theElement = screen.getByRole('textbox', {name: /expiration-date/i});
        userEvent.type(theElement, '01/2022');
        
        expect(theElement).toHaveValue('01/2022');
    });

    it('creates a snapshot', () => {
        const tree = renderer.create(<PaymentInfo />).toJSON();
        expect(tree).toMatchSnapshot();  
    });
}); // end feature

describe('<PaymentInfo /> unit / functional tests', () => {
    beforeEach(() => render(<PaymentInfo />));
    
    it('clicks the finish-order button', () => {
        const theVal = screen.getByRole('button', {name: /finish-order/i});
        fireEvent.click(theVal);
    });
}); // end unit / functional