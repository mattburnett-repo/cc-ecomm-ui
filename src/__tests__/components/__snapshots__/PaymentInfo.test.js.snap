// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentInfo /> feature tests creates a snapshot 1`] = `
<div>
  <div>
    <label
      htmlFor="name-on-card"
    >
      Name on Card:
    </label>
    <input
      aria-label="name-on-card"
      id="name-on-card"
    />
  </div>
  <div>
    <label
      htmlFor="card-number"
    >
      Card Number:
    </label>
    <input
      aria-label="card-number"
      id="card-number"
    />
  </div>
  <div>
    <label
      htmlFor="expiration-date"
    >
      Exp. Date:
    </label>
    <input
      aria-label="expiration-date"
      id="xpiration-date"
    />
  </div>
  <button
    aria-label="finish-order"
  >
    Finish Order
  </button>
</div>
`;
